processor/resourcedetectionprocessor/internal/aws/eks/config.go

```diff
@@ -4,15 +4,28 @@
 package eks // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/aws/eks"
 
 import (
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/aws/eks/internal/metadata"
 )
 
+const imdsCheckMaxRetryDefault = 1
+
 type Config struct {
 	ResourceAttributes metadata.ResourceAttributesConfig `mapstructure:"resource_attributes"`
+	NodeFromEnvVar     string                            `mapstructure:"node_from_env_var"`
+	IMDSCheckMaxRetry  int                               `mapstructure:"imds_check_max_retry"`
+	MaxAttempts        int                               `mapstructure:"max_attempts"`
+	MaxBackoff         time.Duration                     `mapstructure:"max_backoff"`
```

Comment:
Do we really need the additional retries inside the eks detector given that we already have an exponential backoff retry mechanism calling the detector https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/5f2aa3a8dbc2eb6fd3e2f59b37a79b1c54fa805c/processor/resourcedetectionprocessor/internal/resourcedetection.go#L143. Can't we just rely on that instead?

Reply:
But even them, I still think they each have a different purpose; let's say you have code that uses the client multiple times and it's querying a flaky endpoint, the first query succeeds but the second one didn't on the first try, now that we have partial data, you want the client in this case to try again and not count on the detector retry which will start all over.

The `IMDSCheckMaxRetry` is introduced to set the retry to 1 for test access https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/40205/files#diff-2fd609a720e95f7ad6fe8c31180f20671dbf62301fba3c9914fcb81a8cc930aaR388

Finally, this makes it consistent with the ec2 detector that also exposes the same settings.

Reply:
I have seen reports on windows EC2 where the network stack might be slow enough for the imds call to fail.
In this case and if user is confident that they have access to imds endpoint, then they can increase it and adjust the health check settings

Reply:
... my comment about the windows EC2 reports is actually for a standalone EC2 instance, not a k8s windows node, so I guess with k8s windows node, the network stack should be already established .
But yes, if you don't think we might run into an scenario where the we need to increase the retry, then I am fine removing `imds_check_max_retry`

Reply:
Yes, we should prefer not to expose extra options unless they are needed to solve a problem. Excessive configuration interfaces add complexity for users and make maintenance more challenging.

Reply:
@dmitryax 
I removed `imds_check_max_retry` option